tests/network/probes.go

```diff
@@ -107,28 +106,44 @@ var _ = SIGDescribe("[ref_id:1182]Probes", func() {
+		},
+			Entry("[test_id:1202][posneg:positive]with working TCP probe and tcp server on ipv4", tcpProbe, corev1.IPv4Protocol),
+			Entry("[test_id:1202][posneg:positive]with working TCP probe and tcp server on ipv6", tcpProbe, corev1.IPv6Protocol),
+			Entry("[test_id:1200][posneg:positive]with working HTTP probe and http server on ipv4", httpProbe, corev1.IPv4Protocol),
+			Entry("[test_id:1200][posneg:positive]with working HTTP probe and http server on ipv6", httpProbe, corev1.IPv6Protocol),
+			Entry("[test_id:TODO]with working Exec probe", createExecProbe(period, initialSeconds, timeoutSeconds, "uname", "-a"), blankIPFamily),
+		)
+
+		Context("guest agent ping", func() {
+			const (
+				guestAgentConnectTimeout    = 120
+				guestAgentDisconnectTimeout = 300 // Marking the status to not ready can take a little more time
+			)
 
-			if disableEnableCycle {
+			BeforeEach(func() {
+				vmi = tests.NewRandomFedoraVMIWithGuestAgent()
+				vmi.Spec.ReadinessProbe = createGuestAgentPingProbe(period, initialSeconds)
+				vmi = tests.VMILauncherIgnoreWarnings(virtClient)(vmi)
+				By("Waiting for agent to connect")
+				tests.WaitAgentConnected(virtClient, vmi)
+				tests.WaitForVMICondition(virtClient, vmi, v1.VirtualMachineInstanceReady, guestAgentConnectTimeout)
 				By("Disabling the guest-agent")
 				Expect(console.LoginToFedora(vmi)).To(Succeed())
-				Expect(stopGuestAgent(vmi)).ToNot(HaveOccurred())
-
-				// Marking the status to not ready can take a little more time
-				tests.WaitForVMIConditionRemovedOrFalse(virtClient, vmi, v1.VirtualMachineInstanceReady, 300)
-
-				By("Enabling the guest-agent again")
-				Expect(startGuestAgent(vmi)).ToNot(HaveOccurred())
-				tests.WaitForVMICondition(virtClient, vmi, v1.VirtualMachineInstanceReady, 120)
-			}
-		},
-			Entry("[test_id:1202][posneg:positive]with working TCP probe and tcp server on ipv4", tcpProbe, corev1.IPv4Protocol, false),
-			Entry("[test_id:1202][posneg:positive]with working TCP probe and tcp server on ipv6", tcpProbe, corev1.IPv6Protocol, false),
-			Entry("[test_id:1200][posneg:positive]with working HTTP probe and http server on ipv4", httpProbe, corev1.IPv4Protocol, false),
-			Entry("[test_id:1200][posneg:positive]with working HTTP probe and http server on ipv6", httpProbe, corev1.IPv6Protocol, false),
-			Entry("[test_id:TODO]with working Exec probe", createExecProbe(period, initialSeconds, timeoutSeconds, "uname", "-a"), blankIPFamily, false),
-			Entry("[test_id:6739]with GuestAgentPing", guestAgentPingProbe, blankIPFamily, false),
-			Entry("[test_id:6741]status change with guest-agent availability", guestAgentPingProbe, blankIPFamily, true),
-		)
+				Expect(stopGuestAgent(vmi)).To(Succeed())
+				tests.WaitForVMIConditionRemovedOrFalse(virtClient, vmi, v1.VirtualMachineInstanceReady, guestAgentDisconnectTimeout)
+			})
+
+			When("the guest agent is enabled, after being disabled", func() {
```

Comment:
Trying to separate setup and test body  for a single test seems an overkill to me, especially in ginkgo.
The additional indentation is just not worth it IMO, it clutters the test and makes it harder to read.
In this case, it is pretty close, so one may argue it is still ok.

Anyway, I won't insist.

Reply:
Good; I disagree. Having a one liner expectation on the test is preferable, imo.